tests/flytekit/unit/extras/pydantic/test_pydantic_transformer.py

```diff
@@ -0,0 +1,775 @@
+    from dataclasses import dataclass
+    @dataclass
+    class InnerBM:
+        a: int = -1
+        b: float = 3.14
+        c: str = "Hello, Flyte"
+        d: bool = False
+
+    class BM(BaseModel):
+        a: int = -1
+        b: float = 3.14
+        c: str = "Hello, Flyte"
+        d: bool = False
+        inner_bm: InnerBM = Field(default_factory=lambda: InnerBM())
+
+    @task
+    def t_bm(bm: BM):
+        assert isinstance(bm, BM)
+        assert isinstance(bm.inner_bm, InnerBM)
+
+    @task
+    def t_inner(inner_bm: InnerBM):
+        assert isinstance(inner_bm, InnerBM)
+
+    @task
+    def t_test_primitive_attributes(a: int, b: float, c: str, d: bool):
+        assert isinstance(a, int), f"a is not int, it's {type(a)}"
+        assert a == -1
+        assert isinstance(b, float), f"b is not float, it's {type(b)}"
+        assert b == 3.14
+        assert isinstance(c, str), f"c is not str, it's {type(c)}"
+        assert c == "Hello, Flyte"
+        assert isinstance(d, bool), f"d is not bool, it's {type(d)}"
+        assert d is False
+        print("All primitive attributes passed strict type checks.")
+    @workflow
+    def wf(bm: BM):
+        t_bm(bm=bm)
+        t_inner(inner_bm=bm.inner_bm)
+        t_test_primitive_attributes(a=bm.a, b=bm.b, c=bm.c, d=bm.d)
+        t_test_primitive_attributes(a=bm.inner_bm.a, b=bm.inner_bm.b, c=bm.inner_bm.c, d=bm.inner_bm.d)
+
+    bm = BM()
+    wf(bm=bm)
+
+def test_pydantic_dataclasss_in_pydantic_basemodel():
+    from pydantic.dataclasses import dataclass
+    @dataclass
+    class InnerBM:
+        a: int = -1
+        b: float = 3.14
+        c: str = "Hello, Flyte"
+        d: bool = False
+
+    class BM(BaseModel):
+        a: int = -1
+        b: float = 3.14
+        c: str = "Hello, Flyte"
+        d: bool = False
+        inner_bm: InnerBM = Field(default_factory=lambda: InnerBM())
+
+    @task
+    def t_bm(bm: BM):
+        assert isinstance(bm, BM)
+        assert isinstance(bm.inner_bm, InnerBM)
+
+    @task
+    def t_inner(inner_bm: InnerBM):
+        assert isinstance(inner_bm, InnerBM)
+
+    @task
+    def t_test_primitive_attributes(a: int, b: float, c: str, d: bool):
+        assert isinstance(a, int), f"a is not int, it's {type(a)}"
+        assert a == -1
+        assert isinstance(b, float), f"b is not float, it's {type(b)}"
+        assert b == 3.14
+        assert isinstance(c, str), f"c is not str, it's {type(c)}"
+        assert c == "Hello, Flyte"
+        assert isinstance(d, bool), f"d is not bool, it's {type(d)}"
+        assert d is False
+        print("All primitive attributes passed strict type checks.")
+
+    @workflow
+    def wf(bm: BM):
+        t_bm(bm=bm)
+        t_inner(inner_bm=bm.inner_bm)
+        t_test_primitive_attributes(a=bm.a, b=bm.b, c=bm.c, d=bm.d)
+        t_test_primitive_attributes(a=bm.inner_bm.a, b=bm.inner_bm.b, c=bm.inner_bm.c, d=bm.inner_bm.d)
+
+    bm = BM()
+    wf(bm=bm)
+
```

Comment:
can you add two new lines between tests?  I know it doesn't matter, but pycharm complains.